packages/boot/test/orchestration/restart-contracts.test.ts

```diff
@@ -167,3 +167,101 @@ test('stakeAtom', async t => {
   t.is(await flushInboundQueue(), 1);
   t.true(hasResult(wd.getLatestUpdateRecord()));
 });
+
+// Tests restart of an orchestration() flow while an IBC response is pending.
+// TODO make this a macro so we test at arbitrary depth
```

Comment:
this is a little more involved than making a macro. I should reference 9303 in this comment.

it'll need to reset the contract state on each iteration, and since this is a bootstrap test that means either restarting bootstrap or starting a new contract and restarting that one. I guess that works; they can be enumerated. But it will require parameterizing the two builders scripts and the two core-eval functions to parameterize the instance name. I'm not sure it's worth it vs the manual increments being tested here. WDYT @michaelfig ?